### PR TITLE
Add features: Silent Pod and Pod Diagnostics

### PR DESCRIPTION
**Pull request by marionbarker**
This PR adds the Silent Pod and Pod Diagnostics features to the Omnipod screen for DASH pods:

The Silent Pod and Pod Diagnostics rows are added as shown in the graphic below.

User-facing changes:
* Pod notifications include an option to silence the pod
   * When silenced, alerts are only on the phone, the pod does not beep
   * Pod faults, end of life and out of insulin still alarm on the pod regardless of this setting
* The previous pod row is moved under the pod details row to make it easier to find
* A pod diagnostics row is added to the end of the pod screen
   * various diagnostics rows are added in a new sub-screen

The before (left) and after (right) screenshots of the Omnipod DASH screen is shown below.

omnible-pr-117

### CODE REVIEW

File: OmniBLE/PumpManagerUI/Views/OmniBLESettingsView.swift [hunk 1]
```diff
@@ -403,15 +420,27 @@ struct OmniBLESettingsView: View  {
...
                     HStack {
-                        FrameworkLocalText("Confidence Reminders", comment: "Text for confidence reminders navigation link").foregroundColor(Color.primary)
+                        FrameworkLocalText("Confidence Reminders", comment: "Text for confidence reminders navigation link")
+                            .foregroundColor(Color.primary)
                         Spacer()
                         Text(viewModel.beepPreference.title)
                             .foregroundColor(.secondary)
                     }
                 }
+                if  (allowDebugFeatures) {
```
Comment (itsmojo) [inline OmniBLE/PumpManagerUI/Views/OmniBLESettingsView.swift]: Extra space after the "if" keyword.

I'd much prefer that Silence Pod support not be hidden if allowDebugFeatures isn't true as it's a very useful basic feature that should always be available. Also, if it is sometimes hidden and other times not, then if the OmniBLEPumpManagerState.silencePod state variable gets set to true while allowDebugFeatures is true, then this state variable can never be reset by the user and all future pods will never give audible pod alerts if subsequent versions of Loop don't also have allowDebugFeatures set to true.

File: OmniBLE/PumpManagerUI/Views/OmniBLESettingsView.swift [hunk 2]
```diff
@@ -451,14 +480,17 @@ struct OmniBLESettingsView: View  {
                 }
             }
 
-            if let previousPodDetails = viewModel.previousPodDetails {
+            if  (allowDebugFeatures) {
```
Comment (itsmojo) [inline OmniBLE/PumpManagerUI/Views/OmniBLESettingsView.swift]: Extra space after the "if" keyword.

**Review by itsmojo — COMMENTED** [review]
LGTM except for extra spaces in allowDebugFeatures if statements and a big concern about using allowDebugFeatures to conditionally enable displaying the Silence Pod option which I believe should always be available. If this option really must be conditional like this, additional thought and code will be needed to properly handle certain possible cases.

File: OmniBLE/PumpManagerUI/Views/OmniBLESettingsView.swift (hunk 1 above)
Comment (marionbarker) [reply OmniBLE/PumpManagerUI/Views/OmniBLESettingsView.swift]: I was thinking of the requirement needed with LoopKit / MockKit which is also used by Tidepool. But the DIY pump managers will not be shared with Tidepool. So I reverted the commit that added the flag.

**Comment by marionbarker** [comment]
The DIY pump managers will not be used by Tidepool, so I reverted the commit that added the allowDebugFeatures check.

**Comment by marionbarker** [comment]
This graphic illustrates the rows that were present before the PR (left side) compared to the display after the PR (right side).

1. Rename `Device Details` to `Pod Details`
2. Rename `Previous Pod Information` to `Previous Pod Details`
3. Move the `Previous Pod Details` to be directly below `Pod Details`
   * Many users did not know `Previous Pod Information` existed and did not know where to look for Fault Codes when they needed them
   * Enable the display of `Previous Pod Details` even with the first pod, indicating that there is no information using the `-` as opposed to the `>` when there is a previous pod (compare top half of graphic for first pod to bottom half of graphic for second and subsequent pods)
4. Add `Pod Diagnostics` row where `Previous Pod Information` used to be (details for this in a future comment)

new-pod-screen

**Comment by marionbarker** [comment]
This graphic illustrates one of the new Features: Silent Pod
* this is incredibly useful when planning to change a pod in the morning and the user does not want the normal alerts that pod has expired or is about to expire, etc to disturb them while sleeping
* this is also useful for situations when silence is required: during meetings at work, at concerts, in church, etc.
* Critical faults (or out-of-insulin or out-of-time) alerts still sound on the pod itself
* This selection is maintained from pod to pod

pod-silent-screen

**Comment by marionbarker** [comment]
This graphic illustrates the other new feature: Pod Diagnostics

* As we learned with the previous placement of the `Previous Pod Information` row, this is in a locations most users don't notice
* It's available to developers who want this capability
* It's available for trouble-shooting when guided by a mentor
   * Pump manager details can be read easily
   * Eros users can determine the RSSI information for their pod
* Most of the pod diagnostic commands can be run on a faulted pod (before deactivation)
   * The Read Pod Status display will include more detailed info on the Ref codes for a faulted pod

pod-screen-diagnostics

**Review by ps2 — CHANGES_REQUESTED** [review]
`asyncAction` and `FirstAppear` are wrongly pushing view model logic into the view.

**Comment by marionbarker** [comment]
This PR has been updated as requested and is ready for final review.

**Review by ps2 — APPROVED** [review]
LGTM!

**Review by itsmojo — APPROVED** [review]

**Pull request merged**